### PR TITLE
Makes it possible to have more than 16 preferences

### PR DESCRIPTION
**Pull request by Yoshax**
Ports Baystation12/Baystation12#12296

Diff difference is due to:
- Us not having unit tests
- Codebase differences
- Psi's obsession with line tabbing/spacing

### CODE REVIEW

**Comment by PsiOmegaDelta** [comment]
You wish to port https://github.com/Baystation12/Baystation12/pull/12342, https://github.com/Baystation12/Baystation12/pull/12354, and https://github.com/Baystation12/Baystation12/pull/12410 as well.

**Comment by Yoshax** [comment]
I'll add them in shortly.

**Comment by Yoshax** [comment]
Also DNM pending me adding that other stuff and the results of #1217

**Comment by Datraen** [comment]
#1217 Might not go through, depending on the view importance of the crew manifest.

**Comment by Datraen** [comment]
I'd like for those to be cherry picked into the PR as well.

**Comment by Yoshax** [comment]
All done. All 3 of the mentioned PRs are in.

**Comment by Datraen** [comment]
:+1:

**Comment by Yoshax** [comment]
Pleasure ensure the changelog is updated when this is merged, so people are aware their preferences will be reset. It would be a good idea for staff to either make an announcement the first round, or have it in custom event info.

**Comment by Datraen** [comment]
Currently cannot merge, please resolve. Only reason why it was delayed was because of the fact that I didn't want to merge and forget to inform people prior to updating.

**Comment by Yoshax** [comment]
Should be good now, I think.

File: code/modules/admin/verbs/deadsay.dm [hunk 1]
```diff
@@ -8,11 +8,11 @@
...
 		return
 	if(prefs.muted & MUTE_DEADCHAT)
-		src << "\red You cannot send DSAY messages (muted)."
+		src << "<spcan class='warning'>You cannot send DSAY messages (muted).</span>"
 		return
 
-	if(!(prefs.toggles & CHAT_DEAD))
-		src << "\red You have deadchat muted."
+	if(!is_preference_enabled(/datum/client_preference/show_dsay))
+		src << "<spcan class='warning'>You have deadchat muted.</span>"
```
Comment (Datraen) [inline code/modules/admin/verbs/deadsay.dm]: spcan should be span

**Comment by Yoshax** [comment]
Fixed now. Kind of forgot about it.

**Comment by Kelenius** [comment]
Travis disagrees.

File: code/modules/admin/verbs/deadsay.dm [hunk 2]
```diff
@@ -8,11 +8,11 @@
 	if(!src.mob)
 		return
 	if(prefs.muted & MUTE_DEADCHAT)
-		src << "\red You cannot send DSAY messages (muted)."
+		src << "<spcan class='warning'>You cannot send DSAY messages (muted).</span>"
```
Comment (Neerti) [inline code/modules/admin/verbs/deadsay.dm]: Typo here, excessive 'c'.

**Comment by Yoshax** [comment]
My bad, lazily assumed it was the only problem without looking. No idea how those 'c's ended up in there, huh.

**Pull request merged**